### PR TITLE
Manylinux2014

### LINKED ISSUE #4633: reconfigure wheels to build manylinux2014 on master and 2.1
Now that we have dropped 3.6 we should bump to manylinux2014

### PR DESCRIPTION
**Pull request by nabobalis**
Fixes #4633

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -32,59 +32,59 @@ trigger:
     - '*post*'
 
 jobs:
-- template: run-tox-env.yml@OpenAstronomy
```
Comment (nabobalis) [inline azure-pipelines.yml:35]: Undo all this, when the wheels are ok.

**Comment by nabobalis** [comment]
This did not work.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -84,7 +88,7 @@ jobs:
         pytest: false
 
 # On branches which aren't master, and not Pull Requests, build the wheels but only upload them on tags
-- ${{ if and(ne(variables['Build.Reason'], 'PullRequest'), not(contains(variables['Build.SourceBranch'], 'master'))) }}:
+- ${{ if and(ne(variables['Build.Reason'], 'PullRequest'), not(contains(variables['Build.SourceBranch'], 'master'))) or (eq(variables['Build.Reason'], 'Schedule')) }}:
```
Comment (nabobalis) [inline azure-pipelines.yml:91]: It has been a long time since I used azure conditionals.

File: tox.ini [hunk 3]
```diff
@@ -69,7 +69,6 @@ deps =
     # The 32bit build needs numpy installed before trying to install sunpy due
     # to https://github.com/scikit-image/scikit-image/issues/4261
     32bit: numpy
-    32bit: cryptography<3.0
```
Comment (nabobalis) [inline tox.ini:72]: With the update to a functional 32bit image, we no longer need to pin this.

File: azure-pipelines.yml [hunk 4]
```diff
@@ -90,7 +94,7 @@ jobs:
         pytest: false
 
 # On branches which aren't master, and not Pull Requests, build the wheels but only upload them on tags
-- ${{ if and(ne(variables['Build.Reason'], 'PullRequest'), not(contains(variables['Build.SourceBranch'], 'master'))) }}:
+- ${{ if or(ne(variables['Build.Reason'], 'PullRequest'), ne(variables['Build.SourceBranchName'], 'master'), eq(variables['Build.Reason'], 'Schedule'), eq(variables['Build.Reason'], 'Manual')) }}:
```
Comment (Cadair) [inline azure-pipelines.yml:97]: I think you need this:
```suggestion
- ${{ if or(ne(variables['Build.Reason'], 'PullRequest'), or(ne(variables['Build.SourceBranchName'], 'master'), eq(variables['Build.Reason'], 'Schedule'), eq(variables['Build.Reason'], 'Manual'))) }}:
```

**Comment by nabobalis** [comment]
TODO:
Remove py39 wheels
Remove numpy install command
Update changelog
Fix offline helio mock test

File: azure-pipelines.yml [hunk 5]
```diff
@@ -89,8 +81,41 @@ jobs:
         name: python_codestyle
         pytest: false
 
+# Run these only on cron
+- ${{ if or(eq(variables['Build.Reason'], 'Schedule'), eq(variables['Build.Reason'], 'Manual')) }}:
```
Comment (Cadair) [inline azure-pipelines.yml:85]: I think we could not replicate the whole thing. Just put some of the envs list under this conditional.

Comment (nabobalis) [reply azure-pipelines.yml:85]: Good point.

Comment (nabobalis) [reply azure-pipelines.yml:85]: Fixed.

File: azure-pipelines.yml [hunk 6]
```diff
@@ -99,16 +105,13 @@ jobs:
...
-        - wheels_cp3[789]-macosx*
+        - wheels_cp3[78]-manylinux*x86_64
+        - wheels_cp3[78]-macosx*
         - sdist
       dependsOn:
         - py37_test
         - py38_test
         - py39_test
         - py38_test_online
-        - py38_test_32bit
```
Comment (Cadair) [inline azure-pipelines.yml:112]: hmm, perhaps we should change this to allow these to stay? So we run the full lot before a release?

Comment (nabobalis) [reply azure-pipelines.yml:112]: I agree but I am unclear how I would logic this at the moment.

**Review by kakirastern — APPROVED** [review]

File: sunpy/net/helio/tests/test_helio.py [hunk 7]
```diff
@@ -298,10 +298,3 @@ def test_time_query(client):
     table_name = 'rhessi_hxr_flare'
     res = client.time_query(start, end, table=table_name, max_records=10)
     assert len(res.array) == 10
-
-
-def test_client_mock_fail(client):
```
[the remark this answers is not in the document]
Comment (nabobalis) [reply sunpy/net/helio/tests/test_helio.py:303]: It was failing the sdist build, it is also a terrible test.

**Review by dstansby — CHANGES_REQUESTED** [review]
Changes look 👍 , but blocking on adding an action to send a message to sunpy chat for the cron builds.

**Comment by Cadair** [comment]
Just pushed a script for a notification to the chat.

**Review by nabobalis — COMMENTED** [review]
MERGE IT

**Review by dstansby — APPROVED** [review]

**Pull request merged**